marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -315,6 +301,96 @@ function getPrimaryValue(entityInstance, entityDefinition) {
+  let instance = null;
+  let entityTitle = entityInfo.entityName;
+  result.entityProperties = [];
+
+  try {
+    instance = getEntityInstance(result.uri);
+  } catch (error) {
+    console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
+  }
+
+  if (instance != null) {
+    let entityDef = entityInfo.entityModel.definitions[entityTitle];
+    const entityInstance = instance[entityTitle];
+    if (!entityInstance) {
+      console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityTitle}'; will not add entity properties to its search result`);
+    } else {
+      selectedPropertyMetadata.forEach(parentProperty => {
+        result.entityProperties.push(getPropertyValues(parentProperty, entityInstance));
+        result.primaryKey = getPrimaryValue(entityInstance, entityDef);
+      });
+
+      // no primaryKey in entity instance, so use URI
+      if (result.primaryKey === null) {
+        result.primaryKey = {
+          "propertyPath": "uri",
+          "propertyValue": result.uri
+        }
+      }
+    }
+  }
+}
+
+function addGenericEntityProperties(result) {
+  let instance = null;
+  let entityTitle = "";
+  let createdOnDate = "";
+  let entityModel = {};
+  result.primaryKey = {};
+  result.identifier = {};
+  result.entityName = {};
+  result.createdOn = {};
+
+  try {
+    instance = getEntityInstance(result.uri);
+  } catch (error) {
+    console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
+  }
+
+  if (instance != null) {
+    entityTitle = instance.hasOwnProperty("info") ? instance.info.title : Object.keys(instance)[0];
+    entityModel = entityLib.findModelByEntityName(entityTitle);
+    let entityDef = entityModel.definitions[entityTitle];
+    const entityInstance = instance[entityTitle];
+    if (!entityInstance) {
+      console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityTitle}'; will not add entity properties to its search result`);
+    } else {
+
+      result.primaryKey = getPrimaryValue(entityInstance, entityDef);
+
+      // no primaryKey in entity instance, so use URI
+      if (result.primaryKey === null) {
+        result.primaryKey = {
+          "propertyPath": "uri",
+          "propertyValue": result.uri
+        }
+      }
+    }
+  }
+  try {
+    createdOnDate = xdmp.documentGetMetadata(result.uri).datahubCreatedOn;
+  } catch (error) {
+    console.log(`Unable to obtain document with URI '${result.uri}'; will not add createdOn date to its search result`);
+  }
+  let identifierValue = result.primaryKey.propertyPath === "uri" ? result.uri : result.primaryKey.propertyValue;
+  result.identifier = {
+    "propertyPath": "identifier",
+    "propertyValue": identifierValue
+  };
+  result.entityName = {
```

Comment:
Is propertyPath/propertyValue included for entityName and createdOn to make it easier to include in the table? I ask because technically, neither is an entity property, so it's misleading to refer to them as having a propertyPath. So I'd expect to just see "entityName":entityName and "createdOn": createdOnDate. But if including propertyPath/propertyValue makes life simpler for the UI table, that's fine.

Reply:
@rjrudin - Well i added the propertyPath/propertyValue just to be consistent with other fields like primaryKey and Identifier. But we can keep it simple like "entityName":entityName and "createdOn": createdOnDate as it's not like we want it is any specific format for UI. I can make those changes.

Reply:
If it doesn't make life significantly easier for the UI, let's just do "entityName":entityName and likewise for "createdOn" because they're definitely not properties, and thus we shouldn't refer to them having a propertyPath.

Reply:
Done.